cellmlmanip/model.py

```diff
@@ -495,15 +495,24 @@ def get_equations_for(self, symbols, lexicographical_sort=True, recurse=True):
 
         return eqs
 
-    def get_derivative_symbols(self):
+    def get_derivative_symbols(self, order_by_order_added=False):
         """Returns a list of derivative symbols found in the given model graph.
+
+        :param order_by_order_added: indicates whether state_symbols are sorted in the order they appear in the model.
+        Please Note: the order of returned derivative symbols is not guaranteed
```

Comment:
Is this part of the :param: block? Could you change the sentence to put "without the flag" at the start of the sentence and remove the "on Python <= 3.5" block? The contract for the method should be the same regardless of the python version

Reply:
Also, if the options are (1) a particular order, or (2) no particular order, then why not always do (1) ??
So just drop the param and always sort

Reply:
It kind of is part of the param block in a way as otherwise you'd have to refer to the param before it's explained.

I don't mind always sorting, but it's something we should discuss.  I didn't want to change default behavior if I don't have to and also sorting makes it slower in theory (though it's not really noticeable)

Reply:
Yeah I don't think the sorting will slow us down much...

Reply:
What I could to is make the flag default True then if it becomes performance critical there is the option to turn it off.